packages/utils/src/libp2p/index.ts

```diff
@@ -66,21 +68,36 @@ export async function getPeersForProtocol(
   return peers;
 }
 
-export async function getConnectedPeersForProtocol(
+export async function getConnectedPeersForProtocolAndShard(
   connections: Connection[],
   peerStore: PeerStore,
-  protocols: string[]
+  protocols: string[],
+  shardInfo?: ShardInfo
 ): Promise<Peer[]> {
   const openConnections = connections.filter(
     (connection) => connection.status === "open"
   );
 
   const peerPromises = openConnections.map(async (connection) => {
     const peer = await peerStore.get(connection.remotePeer);
-    const supportsProtocol = peer.protocols.some((protocol) =>
-      protocols.includes(protocol)
+    const supportsProtocol = protocols.some((protocol) =>
```

Comment:
nit: `find` will stop at the first occurrence whereas `some` will search through all

Reply:
`find` and `some` both will stop the search as soon as they find a value that satisfies the provided testing function
ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some#description

Reply:
make sense, don't know why I always though otherwise 🤔